### PR TITLE
add progressbar & and fix download attachments

### PR DESCRIPTION
**Pull request by iamdey**
capture d ecran de 2019-01-16 17-07-01

### CODE REVIEW

File: src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx [hunk 1]
```diff
@@ -13,8 +16,104 @@ class MessageAttachments extends Component {
...
+
+  sendFileToBrowser = (index, { data, filename, mime }) => {
+    const blob = new Blob([data], { type: mime || 'application/octet-stream' });
+    const blobURL = window.URL.createObjectURL(blob);
+    const link = document.createElement('a');
+    link.style.display = 'none';
+    link.href = blobURL;
+    link.setAttribute('download', filename);
+    this.downloadRefs[index].ref.current.appendChild(link);
+    link.click();
```
Comment (MonsieurLanza) [inline src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx]: I'm clearly not a fan of this kind of hack… No other way to make DL happen ?

Comment (iamdey) [reply src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx]: I didn't find other way to send the file to the browser. cf. https://github.com/kennethjiang/js-file-download/blob/master/file-download.js

Comment (iamdey) [reply src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx]: … or may be https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/downloads/download
I'll check that

Comment (iamdey) [reply src/frontend/web_application/src/scenes/Discussion/components/MessageAttachments/index.jsx]: arf no, it's an API for WebExtensions

**Pull request merged**